src/Model/Resolver/Products/DataPostProcessor/Attributes.php

```diff
@@ -333,12 +342,37 @@ protected function appendWithSwatchOptions(
         }
     }
 
+    /**
+     * This function collects attribute ids
+     * from catalog_product_super_attribute table
+     * to get which is used for configurable products
+     *
+     * @return array
+     */
+    public function getConfigurableAttributeIds()
+    {
+        $connection = $this->resourceConnection->getConnection();
+        $select = $connection->select();
+
+        $select->from(
+            $connection->getTableName('catalog_product_super_attribute'),
+            'attribute_id'
+        )->distinct(true);
+
+        return $connection->fetchCol($select);
```

Comment:
probably put it in a prop to only be loaded once, should avoid double loading it for variants / regular